store/src/server/api.rs

```diff
@@ -45,6 +48,39 @@ pub async fn serve(
     Ok(())
 }
 
+/// If the database is empty, generates and stores the genesis block. Otherwise, it ensures that the
+/// genesis block in the database is consistent with the genesis block data in the genesis JSON
+/// file.
+async fn ensure_genesis_block(db: &Db) -> Result<()> {
+    let expected_genesis_header: block_header::BlockHeader = {
+        // FIXME: use file path from config (issue #100)
+        let file_contents = fs::read_to_string("TODO FILE PATH")?;
+
+        let genesis_state: GenesisState = serde_json::from_str(&file_contents)?;
+
+        genesis_state.try_into()?
+    };
+
+    let maybe_block_header_in_store = db.select_block_header_by_block_num(Some(0)).await?;
+
+    match maybe_block_header_in_store {
+        Some(block_header) => {
+            // ensure that expected header is what's also in the store
+            if expected_genesis_header != block_header {
+                // TODO: Fill in correct file path
+                return Err(anyhow!("block header in store doesn't match block header in genesis file <TODO FILE PATH>. expected {expected_genesis_header:?}, but store contained {block_header:?}"));
+            }
+        },
+        None => {
+            // add genesis header to store
+
+            // FIXME: `Db::apply_block` requires the channels that are set up in `State`.
+        },
+    }
+
+    Ok(())
+}
```

Comment:
I would probably move this logic into the `Db` struct. Maybe even it could be called as a part of `Db::get_conn()` so that we don't need to call it explicitly. This would allow us to:

1. Call `sql::apply_block()` directly and we don't need to bother with the oneshot channels.
2. We could add `genesis_filepath` to `StoreConfig`.